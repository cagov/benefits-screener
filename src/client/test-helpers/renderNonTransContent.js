import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
