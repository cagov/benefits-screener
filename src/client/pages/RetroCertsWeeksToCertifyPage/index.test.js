import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";
import AUTH_STRINGS from "../../../data/authStrings";

describe("<RetroCertsWeeksToCertifyPage />", () => {
  it("has user data", async () => {
    const wrapper = renderNonTransContent(Component, {
      userData: {
        status: AUTH_STRINGS.statusCode.ok,
        weeksToCertify: [0],
      },
      setUserData: () => {},
    });

    expect(wrapper).toMatchSnapshot();
  });
});
