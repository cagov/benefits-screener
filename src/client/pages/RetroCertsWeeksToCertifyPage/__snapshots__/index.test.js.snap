// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RetroCertsWeeksToCertifyPage /> has user data 1`] = `
<div
  id="overflow-wrapper"
>
  <Header />
  <main
    className="pb-5"
    id="certification-page"
  >
    <div
      className="container p-4"
    >
      <h1>
        Retroactive Certification for Unemployment
      </h1>
      <LanguageSelector
        className="mt-3 mb-4"
      />
      <Alert
        className="mt-5"
        closeLabel="Close alert"
        show={true}
        transition={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "appear": false,
              "in": false,
              "mountOnEnter": false,
              "timeout": 300,
              "unmountOnExit": false,
            },
            "displayName": "Fade",
            "render": [Function],
          }
        }
        variant="success"
      >
        <img
          alt="checkmark"
          className="checkmark"
          src="/images/check-circle-fill.svg"
        />
        You have at least one week available to retroactively certify.
      </Alert>
      <h2
        className="mt-4"
      >
        Important Next Steps
      </h2>
      <p>
        You must retroactively certify for all available weeks.
      </p>
      <ListOfWeeks
        weeksToCertify={
          Array [
            0,
          ]
        }
      />
      <p>
        Answer all retroactive certification questions truthfully and correctly to avoid committing fraud.
      </p>
      <h2>
        Important Information
      </h2>
      <p>
        Your answers for each weekly certification will be saved automatically once you continue to the next week. Partially-completed weeks are not saved. Try to complete all retroactive certifications in one session.
      </p>
      <p>
        To protect the security of your information, this application will automatically time out after 30 minutes of inactivity. To avoid timing out, select any button every 25 minutes.
      </p>
      <Row
        noGutters={false}
      >
        <div
          className="col-md-4"
        >
          <Button
            active={false}
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Link",
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              }
            }
            disabled={false}
            to="/retroactive-certification/certify/2020-02-08"
            type="button"
            variant="secondary"
          >
            Start Certifying
          </Button>
        </div>
      </Row>
    </div>
  </main>
  <Footer
    backToTopTag="certification-page"
  />
</div>
`;
